### DISCUSSION TITLE
Non diff layers

### DISCUSSION THREAD

**Discussion by DanielMadmon** [tracel-ai/burn#1498] [Q&A]
Can I make the optimizer skip custom layers(non-diff layers)?

**Comment by nathanielsimard** [comment]
You can use [Module::no_grad](https://burn.dev/docs/burn/module/trait.Module.html#method.no_grad) on the layers you don't want to optimize.

**Reply by DanielMadmon** [reply]
Is there an example?

**Reply by nathanielsimard** [reply]
I don't think we do at the moment.

**Reply by DanielMadmon** [reply]
ok,thanks maybe I will write one myself to get a better understanding of the framework

**Reply by nathanielsimard** [reply]
That would be really appreciated!
